### PR TITLE
Replace CheckBoxPreference to SwitchPreference

### PR DESCRIPTION
**Pull request by mithlesh4257**
Revamp the Settings Page.
Fix to issue #1039
Replace CheckBoxPreference to SwitchPreference.
Previously discussion #1094
Screenshot:

======

**Titles** (Appearance, General, Feedback) should be at the same size with other font (or a little bigger).
I think it is default.

Give me some links so that I can get for "Separators would be nice between each parts."

### CODE REVIEW

**Comment by maskaravivek** [comment]
@mithlesh4257 Can you rebase your branch with master. :)

**Comment by sivaraam** [comment]
Your commit message in this PR seems to be descriptive but **misleading**. You say that you've replaced CheckBoxPreference to (with?) SwitchPreference but you seem to be doing more than just that. I guess you make multiple *independent* changes at the same time and commit them together. Making multiple independent changes at one time is not a big issue, it happens most of the time for me. But committing them together and giving it a misleading commit message is a *big issue*. You're making it difficult for others to find who did a particular change. For example, In case somebody would like to know who made the categorization to the settings page by searching the `git log` output your commit message is not helpful.

In case you like to know how to write good commit messages see,

* [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)

* "Commit Guidelines" sub-section of the [Distributed Git - Contributing to a Project](https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project) section in the ProGit2 book.

For an example of good commit messages you could see the [history of this feature branch for Git](https://github.com/gitster/git/commits/sb/diff-blobfind-pickaxe). In general `git log --no-merges` on history of [Git](https://github.com/git/git) would give you an idea of how good commit message look. Their [SubmittingPatches](https://github.com/git/git/blob/master/Documentation/SubmittingPatches) document is also a good read for contributors.

**Comment by mithlesh4257** [comment]
Hi @maskaravivek :) how to rebase branch with master.

**Comment by maskaravivek** [comment]
@mithlesh4257, in the comment above @sivaraam has already given an overview of how to use Git. We expect a basic familiarity with git before you contribute to this project. 

It would be great if things which a quick google search would answer are not discussed in a PR. Please make a honest effort of trying it yourself before asking about git workflow. :)

**Comment by mithlesh4257** [comment]
@maskaravivek, I have gone through [Rebse local branch to remote master](https://stackoverflow.com/questions/7929369/how-to-rebase-local-branch-with-remote-master). I perfomed all the correct steps.

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -217,9 +217,7 @@
...
   <string name="read_article">READ ARTICLE</string>
-
-  <string name="notifications_welcome" formatted="false">Welcome to Wikimedia Commons, %s! We\'re glad you\'re here.</string>
-  <string name="notifications_talk_page_message" formatted="false">%s left a message on your talk page</string>
-  <string name="notifications_thank_you_edit">Thank you for making an edit</string>
-  <string name="notifications_mention" formatted="false">%s mentioned you on %s.</string>
+  <string name="appearance">Appearance</string>
+  <string name="general">General</string>
+  <string name="feedback">Feedback</string>
 </resources>
```
Comment (neslihanturan) [inline app/src/main/res/values/strings.xml]: Why these lines are deleted?

Comment (mithlesh4257) [reply app/src/main/res/values/strings.xml]: Let me see it and I would change it.

**Comment by neslihanturan** [comment]
This PR has also conflicts, you need to solve them so that I can merge it after reviews are done.

**Comment by neslihanturan** [comment]
Please re-open it if you decide to solve conflicts:)

**Comment by sivaraam** [comment]
> Please re-open it if you decide to solve conflicts:)

I think there's already another PR that fixed the same issue that this one tries to address: #1259

**Comment by neslihanturan** [comment]
Oh, thanks for remind @sivaraam . Our merge cycle is very fast todays. So, yes we can't re-open this one.

**Pull request closed without merging**